app/forms/candidate_interface/continuous_applications/which_course_are_you_applying_to_step.rb

```diff
@@ -24,25 +26,35 @@ def previous_step
       def next_step
         return :course_review if completed?
 
-        if multiple_study_modes?
+        if duplicate_course?
+          :duplicate_course_selection
+        elsif multiple_study_modes?
           :course_study_mode
         elsif multiple_sites?
           :course_site
         end
       end
 
+      def next_edit_step_path(next_step_klass)
+        return '' if next_step_klass == DuplicateCourseSelectionStep
+
+        super
+      end
+
       def next_step_path_arguments
         if completed?
           default_path_arguments
-        elsif multiple_study_modes?
+        elsif duplicate_course? || multiple_study_modes?
           { provider_id:, course_id: }
         elsif multiple_sites?
           { provider_id:, course_id:, study_mode: }
         end
       end
 
       def edit_next_step_path_arguments
-        if completed?
+        if duplicate_course?
+          default_path_arguments.merge({ provider_id:, course_id: })
```

Comment:
`default_path_arguments.merge(provider_id:, course_id:)` No need for curly brackets, but does this code ever get called?